src/spec.h

```diff
@@ -82,6 +82,11 @@ typedef struct fieldSpec {
 
   int sortIdx;
 
+  /**
+   * Unique field index. Each field has a unique index regardless of its type
+   */
+  uint16_t index;
```

Comment:
I know we said `int`, lmk if this needs to be changed, I forgot to change it back.